src/internal/operations/backup_test.go

```diff
@@ -31,7 +29,272 @@ import (
+
+	if r == nil {
+		return nil, errors.Errorf("no details for ID %s", detailsID)
+	}
+
+	return r, nil
+}
+
+// ----- model store for backups
+
+type mockBackupStorer struct {
+	// Only using this to store backup models right now.
+	entries map[model.StableID]backup.Backup
+}
+
+func (mbs mockBackupStorer) Get(
+	ctx context.Context,
+	s model.Schema,
+	id model.StableID,
+	toPopulate model.Model,
+) error {
+	if s != model.BackupSchema {
+		return errors.Errorf("unexpected schema %s", s)
+	}
+
+	r, ok := mbs.entries[id]
+	if !ok {
+		return errors.Errorf("model with id %s not found", id)
+	}
+
+	bu, ok := toPopulate.(*backup.Backup)
+	if !ok {
+		return errors.Errorf("bad input type %T", toPopulate)
+	}
+
+	*bu = r
+
+	return nil
+}
+
+func (mbs mockBackupStorer) Delete(context.Context, model.Schema, model.StableID) error {
+	return errors.New("not implemented")
+}
+
+func (mbs mockBackupStorer) DeleteWithModelStoreID(context.Context, manifest.ID) error {
+	return errors.New("not implemented")
+}
+
+func (mbs mockBackupStorer) GetIDsForType(
+	context.Context,
+	model.Schema,
+	map[string]string,
+) ([]*model.BaseModel, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (mbs mockBackupStorer) GetWithModelStoreID(
+	context.Context,
+	model.Schema,
+	manifest.ID,
+	model.Model,
+) error {
+	return errors.New("not implemented")
+}
+
+func (mbs mockBackupStorer) Put(context.Context, model.Schema, model.Model) error {
+	return errors.New("not implemented")
+}
+
+func (mbs mockBackupStorer) Update(context.Context, model.Schema, model.Model) error {
+	return errors.New("not implemented")
+}
+
+// ---------------------------------------------------------------------------
+// helper funcs
+// ---------------------------------------------------------------------------
+
+func makeMetadataPath(
+	t *testing.T,
+	tenant string,
+	service path.ServiceType,
+	resourceOwner string,
+	category path.CategoryType,
+	fileName string,
+) path.Path {
+	t.Helper()
+
+	p, err := path.Builder{}.Append(fileName).ToServiceCategoryMetadataPath(
+		tenant,
+		resourceOwner,
+		service,
+		category,
+		true,
+	)
+	require.NoError(t, err)
+
+	return p
+}
+
+func makeFolderEntry(
+	t *testing.T,
+	pb *path.Builder,
+	size int64,
+	modTime time.Time,
+) *details.DetailsEntry {
+	t.Helper()
+
+	return &details.DetailsEntry{
+		RepoRef:   pb.String(),
+		ShortRef:  pb.ShortRef(),
+		ParentRef: pb.Dir().ShortRef(),
+		ItemInfo: details.ItemInfo{
+			Folder: &details.FolderInfo{
+				ItemType:    details.FolderItem,
+				DisplayName: pb.Elements()[len(pb.Elements())-1],
+				Modified:    modTime,
+				Size:        size,
+			},
+		},
+	}
+}
+
+// TODO(ashmrtn): Really need to factor a function like this out into some
+// common file that is only compiled for tests.
+func makePath(t *testing.T, elements []string, isItem bool) path.Path {
+	t.Helper()
+
+	p, err := path.FromDataLayerPath(stdpath.Join(elements...), isItem)
+	require.NoError(t, err)
+
+	return p
+}
```

Comment:
Why would we not place this in the `testing` package?

Reply:
Circular dependencies, mostly, but also smells.  /Testing, in general, shouldn't produce higher level package's structs, else those packages are unable to use the /testing package due to circular imports.